Libraries/oneMKL/student_t_test/t_test.cpp

```diff
@@ -0,0 +1,201 @@
+//==============================================================
+// Copyright © 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+/*
+ *
+ *  Content:
+ *       This file contains Student's T-test DPC++ implementation with
+ *       buffer APIs.
+ *
+ *******************************************************************************/
+
+#include <CL/sycl.hpp>
+#include <cstdlib>
+#include <iostream>
+#include <vector>
+
+#include "oneapi/mkl.hpp"
+
+using fp_type = float;
+// Initialization value for random number generator
+static const auto seed = 7777;
+// Quantity of samples to check using Students' T-test
+static const auto n_samples = 1000000;
+// Expected mean value of random samples
+static const auto expected_mean = 0.0f;
+// Expected standard deviation of random samples
+static const auto expected_std_dev = 1.0f;
+// T-test threshold which corresponds to 5% significance level and infinite
+// degrees of freedom
+static const auto threshold = 1.95996f;
+
+// T-test function with expected mean
+// Returns: -1 if something went wrong, 1 - in case of NULL hypothesis should be
+// accepted, 0 - in case of NULL hypothesis should be rejected
+template <typename RealType>
+std::int32_t t_test(sycl::queue& queue, sycl::buffer<RealType, 1>& r,

```

Comment:
Does it help to shorten "queue" to "q"  as the variable?

Reply:
changed, though it doesn't lessen quantity of lines